app/access/space_quota_definition_access.rb

```diff
@@ -70,7 +70,7 @@ def delete?(space_quota_definition, params=nil)
     def read?(space_quota_definition, *_)
       context.admin_override || (
         !context.user.nil? && (
-          (context.user.managed_organizations.include? space_quota_definition.organization) ||
+          context.user.managed_organizations_dataset.where(id: space_quota_definition.organization_id).any? ||
           !(context.user.managed_spaces & space_quota_definition.spaces).empty? ||
```

Comment:
This is definitely an improvement, but you can go further and cut out organizations table by doing something like

`OrganizationManager.where(user_id: context.user.id, organization_id: space_quota_definition.organization.id).any?`

Reply:
`managed organizations_dataset` produces this query:

```
SELECT  "organizations".*
FROM    "organizations"
       INNER JOIN  "organizations_managers"
               ON (  "organizations_managers" . "organization_id" = 
                    "organizations" . "id" )
WHERE  (  "organizations_managers" . "user_id" = 118 ) 
```